libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/route/RouteFetcher.java

```diff
@@ -152,16 +151,36 @@ private String[] calculateRemainingApproaches(RouteProgress routeProgress) {
     return approaches;
   }
 
-  private void executeRouteCall(NavigationRoute.Builder builder) {
+  /**
+   * Cancels the Directions API call if it has not been executed yet.
+   */
+  public void cancelRouteCall() {
+    if (navigationRoute != null) {
+      navigationRoute.cancelCall();
+    }
+  }
+
+  /**
+   * Executes the given NavigationRoute builder, eventually triggering
+   * any {@link RouteListener} that has been added via {@link RouteFetcher#addRouteListener(RouteListener)}.
+   *
+   * @param builder to be executed
+   */
+  public void findRouteWith(NavigationRoute.Builder builder) {
     if (builder != null) {
-      builder.build().getRoute(directionsResponseCallback);
+      navigationRoute = builder.build();
+      navigationRoute.getRoute(directionsResponseCallback);
     }
   }
 
   private boolean isInvalidProgress(Location location, RouteProgress routeProgress) {
     return location == null || routeProgress == null;
   }
 
+  private boolean invalid(Context context, Location location, RouteProgress routeProgress) {
```

Comment:
Isn't this pretty much a copy of isInvalidProgress?

Reply:
I was adapting those versions to the latest mapbox version. Since this version and adaptaions already had some bug fix iterations. So is adapted those changes in the RouteFetcher file. Also some former private methods where needed in various places in the navigation-ui module

Reply:
@wipfli this code from the tag i used as a base was published under MIT: https://github.com/mapbox/mapbox-navigation-android/tree/v0.42.6